Compile issue on Clang 19.1.6

Occurring on [CI](https://github.com/moves-rwth/storm/actions/runs/12762230745/job/35570433490) with latest Clang on Archlinux.

```
[ 60%] Building CXX object src/storm/CMakeFiles/storm.dir/generator/StateBehavior.cpp.o
/opt/storm/src/storm/generator/JaniNextStateGenerator.cpp:310:32: error: a template argument list is expected after a name prefixed by the template keyword [-Wmissing-template-arg-list-after-template-kw]
310 |             allValues.template emplace_back(aInitLocs.begin(), aInitLocs.end());
    |                                ^
```

--- a/src/storm/generator/JaniNextStateGenerator.cpp
+++ b/src/storm/generator/JaniNextStateGenerator.cpp
@@ -307,7 +307,7 @@ std::vector<StateType> JaniNextStateGenerator<ValueType, StateType>::getInitialS
         std::vector<std::vector<uint64_t>> allValues;
         for (auto const& aRef : this->parallelAutomata) {
             auto const& aInitLocs = aRef.get().getInitialLocationIndices();
-            allValues.template emplace_back(aInitLocs.begin(), aInitLocs.end());
+            allValues.emplace_back(aInitLocs.begin(), aInitLocs.end());
         }
         uint64_t locEndIndex = allValues.size();
         for (auto const& intVar : this->variableInformation.integerVariables) {

--- a/src/storm/solver/helper/IntervalterationHelper.cpp
+++ b/src/storm/solver/helper/IntervalterationHelper.cpp
@@ -105,7 +105,7 @@ SolverStatus IntervalIterationHelper<ValueType, TrivialRowGrouping>::II(std::pai
     }
     while (status == SolverStatus::InProgress) {
         ++numIterations;
-        viOperator->template applyInPlace(xy, offsets, backend);
+        viOperator->applyInPlace(xy, offsets, backend);
         if (checkConvergence(xy, convergenceCheckState, getNextConvergenceCheckState, relative, precision)) {
             status = SolverStatus::Converged;
         } else if (iterationCallback) {

--- a/src/storm/solver/helper/OptimisticValueIterationHelper.cpp
+++ b/src/storm/solver/helper/OptimisticValueIterationHelper.cpp
@@ -73,7 +73,7 @@ SolverStatus OptimisticValueIterationHelper<ValueType, TrivialRowGrouping>::GSVI
     SolverStatus status{SolverStatus::InProgress};
     while (status == SolverStatus::InProgress) {
         ++numIterations;
-        if (viOperator->template applyInPlace(operand, offsets, backend)) {
+        if (viOperator->applyInPlace(operand, offsets, backend)) {
             status = SolverStatus::Converged;
         } else if (iterationCallback) {
             status = iterationCallback(status, operand);
@@ -210,7 +210,7 @@ SolverStatus OptimisticValueIterationHelper<ValueType, TrivialRowGrouping>::OVI(
         }
         while (numIterations < maxIters) {
             ++numIterations;
-            if (viOperator->template applyInPlace(vu, offsets, backend)) {
+            if (viOperator->applyInPlace(vu, offsets, backend)) {
                 if (backend.allDown()) {
                     return SolverStatus::Converged;
                 } else {

--- a/src/storm/solver/helper/SoundValueIterationHelper.cpp
+++ b/src/storm/solver/helper/SoundValueIterationHelper.cpp
@@ -331,7 +331,7 @@ typename SoundValueIterationHelper<ValueType, TrivialRowGrouping>::SVIData Sound
 
     while (true) {
         ++numIterations;
-        viOperator->template applyInPlace(xy, offsets, backend);
+        viOperator->applyInPlace(xy, offsets, backend);
         SVIData data{SolverStatus::InProgress, xy, backend.a(), backend.b()};
         if (data.checkConvergence(convergenceCheckState, getNextConvergenceCheckState, relative, precision)) {
             return SVIData{SolverStatus::Converged, xy, backend.a(), backend.b()};

